src/gfn/gflownet/base.py

```diff
@@ -70,11 +69,26 @@ def to_training_samples(self, trajectories: Trajectories) -> TrainingSampleType:
         """Converts trajectories to training samples. The type depends on the GFlowNet."""
 
     @abstractmethod
-    def loss(self, env: Env, training_objects: Any) -> torch.Tensor:
-        """Computes the loss given the training objects."""
+    def loss(
+        self,
+        env: Env,
+        training_objects: Any,
+        recalculate_all_logprobs: bool = True,
+    ) -> torch.Tensor:
+        """Computes the loss given the training objects.
+
+        Args:
+            env: The environment to compute the loss for
+            training_objects: The objects to compute the loss on
+            recalculate_all_logprobs: If True, always recalculate logprobs even
+                if they exist. If False, use existing logprobs when available.
+        """
 
     def loss_from_trajectories(
-        self, env: Env, trajectories: Trajectories
+        self,
+        env: Env,
+        trajectories: Trajectories,
+        recalculate_all_logprobs: bool = True,
```

Comment:
Given that this is the default behavior now, should we change anything in the scripts?

Reply:
Perhaps yes - I haven't looked at it yet.

Reply:
Actually yes, it's important. Both in scripts and tests.
I'm on it...